acoular/loudness.py

```diff
@@ -0,0 +1,215 @@
+                                          self.sample_freq, 
+                                          field_type=self.field_type)[0:3]
+                self.overall_loudness[i] = N
+                self.specific_loudness[:,i] = N_spec
+
+        else:   
+            self.overall_loudness, self.specific_loudness, self.bark_axis = \
+                loudness_zwst(self._time_data[:], self.sample_freq, 
+                              field_type=self.field_type)[0:3]
+            
+
+class LoudnessTimevariant(_Loudness):
+    """
+    Calculates the timevariant loudness according to ISO 532-1 (Zwicker) 
+    from a given source. Calculates loudness for timesteps of 64 samples.
+
+    Uses third party code from `mosqito 
+    <https://mosqito.readthedocs.io/en/latest/index.html>`__.
+
+    Parameters
+    ==========
+    source : SamplesGenerator or derived object?
+        The input source as TimeSamples object.
+
+    References
+    ==========
+    - Acoustics –
+      Methods for calculating loudness –
+      Part 1: Zwicker method (ISO 532-1:2017, Corrected version 2017-11)
+    - Green Forge Coop. (2024). MOSQITO (Version 1.2.1) [Computer software]. 
+      https://doi.org/10.5281/zenodo.11026796
+    """
+    overall_loudness = Union(CArray(),
+                             desc="overall loudness (shape: `N_channels x" 
+                             " N_times)")
+
+    specific_loudness = CArray(desc="specific loudness sones/bark per channel "
+                               "(shape: `N_channels x N_bark x N_times`).")
+
+    
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    @observe('source', post_init=True)
+    def _source_changed(self, event):
+        """
+        Observer method that is called whenever the `source` attribute changes.
+        Fetches time data via result() in blocks of size `_block_size`.
+        """
+        self._time_data = np.empty((self.numsamples, self.numchannels))
+        i = 0
+
+        for res in self.source.result(self._block_size):
+            n_samples = res.shape[0]
+            self._time_data[i : i + n_samples] = res
+            i += n_samples
+
+        self._calculate_loudness()
+
+    def _calculate_loudness(self):
+        """
+        Private function to calculate overall and specific loudness.
+        """
+        print("Calculating timevariant loudness... depending on the file size, " 
+              "this might take a while")
+        
+        # resample to 48 kHz
+        if self.sample_freq != 48000:
+            self._resample_to_48kHz()
+
+        # get ntime, code from mosqito
+        dec_factor = int(self.sample_freq / 2000)
+        n_time = int(len(self._time_data[:,0][::dec_factor]) / 4)
+
+        self.overall_loudness = np.zeros((self.numchannels, n_time))
+        self.specific_loudness = np.zeros((self.numchannels, 240, n_time)) # restructure plot code to bark x channels x time as in stationary loudness?
```

Comment:
@slis-sys 
Die specific loudness wird in stationary als `n_bark x n_channels `zurückgegeben
in timevariant in `n_channels x n_bark x n_time`. Wollen wir das einheitlich machen? Also z.B. `n_bark x n_channels x n_time`
Müsste dann in der Plotklasse angepasst werden.